hphp/hack/src/parser/full_fidelity_ast.ml

```diff
@@ -3587,6 +3591,7 @@ if there already is one, since that one will likely be better than this one. *)
         let is_abs = List.mem kinds Abstract ~equal:( = ) in
         match token_kind kw with
         | Some TK.Class when is_abs -> Cabstract
+        | Some TK.Class when c_is_xhp -> Cxhp
```

Comment:
What about `abstract xhp class Foo {}`?

Reply:
Good point. I added a case for when a class is abstract and xhp to make it xhp.